TWLight/users/oauth.py

```diff
@@ -264,10 +265,11 @@ def authenticate(self, request=None, access_token=None, handshaker=None):
         return user
 
     def get_user(self, user_id):
-        try:
-            return User.objects.get(pk=user_id)
-        except User.DoesNotExist as e:
-            logger.exception(e)
+        user = User.objects.filter(pk=user_id).first()
+        if user is not None:
+            return user
+        else:
+            logger.warning("OAuthBackend.get_user: User does not exist")
             return None
```

Comment:
I had to go lookup the django docs for how this method is supposed to work.
https://docs.djangoproject.com/en/4.2/ref/contrib/auth/#django.contrib.auth.get_user

I'm pretty sure I created the no return problem here years ago. We should probably document in this class that it's implementing `django.contrib.auth`, just as a reminder to ourselves.
suggestion for legibility
```suggestion
        user = User.objects.filter(pk=user_id).first()
        if user is None:
            logger.warning("OAuthBackend.get_user: User does not exist")
        return user         
```
```suggestion
        user = User.objects.filter(pk=user_id).first()
        if user is not None:
            return user
        else:
            logger.warning("OAuthBackend.get_user: User does not exist")
            return None
```

Reply:
Gotcha! I changed the logger from `exception` to `warning` because it was throwing errors when it shouldn't have and it is making noise in our GlitchTip.